build.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+# cugraph build script
+
+# This script is used to build the component(s) in this repo from
+# source, and can be called with various options to customize the
+# build as needed (see the help output for details)
+
+# Abort script on first error
+set -e
+
+NUMARGS=$#
+ARGS=$*
+
+# NOTE: ensure all dir changes are relative to the location of this
+# script, and that this script resides in the repo dir!
+REPODIR=$(cd $(dirname $0); pwd)
+
+VALIDARGS="clean libcugraph cugraph -v -g -n -h --help"
+HELP="$0 [<target> ...] [<flag> ...]
+ where <target> is:
+   clean      - remove all existing build artifacts and configuration (start over)
+   libcugraph - build the cugraph C++ code
+   cugraph    - build the cugraph Python package
+ and <flag> is:
+   -v         - verbose build mode
+   -g         - build for debug
+   -n         - no install step
+   -h         - print this text
+
+ default action (no args) is to build and install 'libcugraph' then 'cugraph' targets
+"
+LIBCUGRAPH_BUILD_DIR=${REPODIR}/cpp/build
+CUGRAPH_BUILD_DIR=${REPODIR}/python/build
+BUILD_DIRS="${LIBCUGRAPH_BUILD_DIR} ${CUGRAPH_BUILD_DIR}"
+
+# Set defaults for vars modified by flags to this script
+VERBOSE=""
+BUILD_TYPE=Release
+INSTALL_TARGET=install
+
+# Set defaults for vars that may not have been defined externally
+#  FIXME: if INSTALL_PREFIX is not set, check PREFIX, then check
+#         CONDA_PREFIX, but there is no fallback from there!
```

Comment:
When one would use `PREFIX` or `INSTALL_PREFIX` instead of `CONDA_PREFIX`? 
cugraph devs use `CONDA_PREFIX` almost exclusively. Should  `CONDA_PREFIX`  be the default? It seems to me that cugraph may not be the only product interested in setting `PREFIX` or `INSTALL_PREFIX` env variables. That may lead to an unwanted install location if they are preferred to  `CONDA_PREFIX` .

Reply:
conda-build uses $PREFIX in the recipes so we still need to support that, but I believe I can safely change the precedence (it can check CONDA_PREFIX first, then PREFIX). INSTALL_PREFIX is a more self-documenting alternative being offered here, but I can remove that one altogether.

Reply:
after our in-person discussion, I'll remove `INSTALL_PREFIX` but leave the precedence of `PREFIX` over `CONDA_PREFIX` as-is to allow someone to override conda's location if necessary.